src/transformers/generation/utils.py

```diff
@@ -435,7 +438,10 @@ def prepare_inputs_for_generation(
             and position_ids_key in set(inspect.signature(self.forward).parameters.keys())
         ):
             position_ids = attention_mask.long().cumsum(-1) - 1
-            position_ids.masked_fill_(attention_mask == 0, 1)
+            if position_ids.device.type == "hpu":
+                position_ids[attention_mask == 0] = 1
+            else:
+                position_ids.masked_fill_(attention_mask == 0, 1)
```

Comment:
same as above.

Reply:
It seems this fix is not needed on the new version of SynapseAI 1.20.0 + Pytorch 2.6, so we'll probably target these two as minimum requirement https://huggingface.slack.com/archives/C03264192KV/p1741272959824279?thread_ts=1739977452.256159&cid=C03264192KV.